handlers/handler.go

```diff
@@ -95,6 +99,30 @@ func (env *Env) PostUserHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(reqUser)
 }
 
+func (env *Env) PatchUserHandler(w http.ResponseWriter, r *http.Request) {
+	reqUser := &models.User{}
+	if err := parseJSON(w, r.Body, reqUser); err != nil {
+		return
+	}
+	if reqUser.Email == "" ||
+		(reqUser.Name == "" && reqUser.Password == "" && reqUser.AvatarURL == "") {
```

Comment:
I think `Email` should be the only required field and if none of the other fields are provided, just respond with a `200 OK` and the `reqUser`.